internal/pkg/mapper/mapper.go

```diff
@@ -0,0 +1,69 @@
+package mapper
+
+import (
+	"go.uber.org/zap"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/filesystem"
+	"github.com/keboola/keboola-as-code/internal/pkg/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/transformation"
+)
+
+type state = model.State
+type Mapper struct {
+	*state
+	logger *zap.SugaredLogger
+	fs     filesystem.Fs
+	naming model.Naming
+}
+
+func New(state *model.State, logger *zap.SugaredLogger, fs filesystem.Fs, naming model.Naming) *Mapper {
+	return &Mapper{state: state, logger: logger, fs: fs, naming: naming}
+}
+
+func (m *Mapper) isTransformationConfig(object interface{}) (bool, error) {
+	v, ok := object.(*model.Config)
+	if !ok {
+		return false, nil
+	}
+
+	component, err := m.Components().Get(*v.ComponentKey())
+	if err != nil {
+		return false, err
+	}
+
+	return component.IsTransformation(), nil
+}
+
+func (m *Mapper) BeforeSave(files *model.ObjectFiles) error {
```

Comment:
Mapper ma metody `BeforeSave` a `AfterLoad`, ktore umoznuju upravit podobu objektu na disku/v pamati, ked sa uklada/nacitava.

To sa pouzije aj pre dalsie features, ... teraz primarne pre `native shared codes`.

Narozdiel od predchadzajuceho kodu, na vstupe je `model.ObjectFiles`, tj. je mozne lubovolne menit vsetky subory viazane na objet a aj pridavat nove.

Staci ich zapisat do `files` a automaticky sa zapisu aj na disk (vid `local/save.go`).

Reply:
Najlepsie je pozriet asi finalne subory:
- https://github.com/keboola/keboola-as-code/blob/c1944bd24a6e4fcbd2f4c68d260869c9746de8b1/internal/pkg/transformation/load.go
- https://github.com/keboola/keboola-as-code/blob/c1944bd24a6e4fcbd2f4c68d260869c9746de8b1/internal/pkg/transformation/save.go